core/arch/arm/plat-imx/imx_pl310.c

```diff
@@ -68,13 +68,6 @@ void arm_cl2_enable(vaddr_t pl310_base)
 
 vaddr_t pl310_base(void)
 {
-	static void *va;
-
-	if (cpu_mmu_enabled()) {
-		if (!va)
-			va = phys_to_virt(PL310_BASE, MEM_AREA_IO_SEC);
-		return (vaddr_t)va;
-	}
-	return PL310_BASE;
+	return core_mmu_get_io_va(PL310_BASE);
```

Comment:
Another point. previous code used a local static storage for the va. Hence conversion is done only once.
With this change, `phys_to_virt_io()` is called each time pl310 is used.